dataservice/api/common/validation.py

```diff
@@ -47,3 +47,54 @@ def enum_validation_generator(_enum, common=True):
         ', '.join([str(el) for el in extended_enum]))
 
     return OneOf(extended_enum, error=error_message)
+
+
+def list_validation_generator(valid_items, items_name='items'):
+    """
+    Return a list validation function
+
+    :param valid_items: valid collection of items to use in validation
+    :type valid_items: list
+    :items_name: the plural identifier of an item in the list. Used in the
+    ValidationError message
+
+    :returns: the validate_list method
+    """
+    def validate_list(input_items):
+        """
+        Check whether all items in the input list exist in another list
+        representing the collection of valid values.
+
+        All list items are converted to strings before
+        doing comparison and validation.
+
+        ** NOTE **
+        This method is used by marshmallow_sqlalchemy.field_for during
+        HTTP request validation. If the HTTP request method is GET,
+        `input_items` will be a list with one element which will be a delimited
+        string representing a list of strings
+        (e.g. ['duo:0000005,duo:0000001']). This is because the value comes
+        from the URL query parameter string. No need to validate this case.
+
+        :param input_items: input collection of items to validate
+        :type input_items: list or str
+        :raises: ValidationError when validation fails
+        """
+        assert isinstance(input_items, list), (
+            'Parameter `input_items` must be a list'
+        )
+        # Don't do validation for this case, just return. See NOTE in docstring
+        if len(input_items) == 1 and any([d in input_items[0]
+                                          for d in [',', ' ', ';']]):
+            return
+
+        valid_set = {str(valid_item) for valid_item in valid_items}
+        invalid_set = {item for item in input_items
+                       if str(item) not in valid_set}
```

Comment:
I think you can avoid the comprehensions here and reduce to one line:
```
invalid_set = set(input_items) - set(valid_items)
```

Reply:
Ok 👍 I implemented ^ bc at first I was doing case insensitive validation. I've decided against doing case insensitive validation (see reason in [comment](https://github.com/kids-first/kf-api-dataservice/pull/543#discussion_r340392528) earlier) so this should be fine